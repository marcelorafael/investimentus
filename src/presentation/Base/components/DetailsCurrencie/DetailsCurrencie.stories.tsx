import React from 'react'
import { text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react-native'

import { DetailsCurrencie } from './DetailsCurrencie'
import { View } from 'react-native'

storiesOf('Base/DetailsCurrencie', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <View style={{ flex: 1, alignItems: 'center', padding: 10 }}>
      <DetailsCurrencie
        company_name={text('companyName', 'Banco Inter - S.A')}
        price={text('price', '120,00')}
        description={text(
          'description',
          'Texto de descrição sobre a desterminada ação a qual estamos pesquisando'
        )}
        currency={text('currency', 'BRL')}
        region={text('region', 'Saõ-Paulo')}
        close="12:00"
        open={text('open', '10:00')}
        change_percert={text('change_percent', '-10')}
        name={text('name', 'Inter')}
        symbol={text('symbol', 'BIDI4')}
        market_cap={text('market', '2222.0')}
      />
    </View>
  ))
