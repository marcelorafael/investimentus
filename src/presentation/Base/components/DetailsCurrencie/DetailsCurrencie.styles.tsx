import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';

export const Container = styled(LinearGradient).attrs({
  colors:['#000','#7F5DF0','#000']
})`

width: 300px;
max-width: 400px;
max-height: 500px;

padding-top: 7px;
border-radius: ${props => props?.theme?.SIZES?.radius ?? 5}px;
`;

export const CenterView = styled.View`
padding-left: 20px;
padding-right: 20px;
`;

export const Title = styled.Text`
${props => props?.theme?.FONTS?.oswaldBold ?? {}}
color: ${props => props?.theme?.COLORS?.white ?? '#FFF'};
text-align: center;
`;

export const Description = styled.Text`
${props => props?.theme?.FONTS?.oswald ?? {}}
color: ${props => props?.theme?.COLORS?.white ?? '#FFF'};
`;

export const DescriptionTitle = styled.Text`
${props => props?.theme?.FONTS?.robotoThin ?? {}}
color: ${props => props?.theme?.COLORS?.white ?? '#FFF'};
`;

export const Footer = styled.View`
background-color: ${props => props?.theme?.COLORS?.lightGray1 ?? '#FFF'};
width: 100%;
min-height: 80px;
max-height: 200px;

border-bottom-left-radius: 5px;
border-bottom-right-radius: 5px;
padding: 5px;
`;

export const DescriptionFooter = styled.Text`
${props => props?.theme?.FONTS?.oswald ?? {}}
color: ${props => props?.theme?.COLORS?.gray ?? '#FFF'};
`;

export const WrapperDescriptions = styled.View`
flex-direction: row;
justify-content: space-between;
margin-bottom: 7px;
`;