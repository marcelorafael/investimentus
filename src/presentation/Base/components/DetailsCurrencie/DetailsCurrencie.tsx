import React from 'react'
import { StyleSheet, View } from 'react-native'
import {
  Container,
  Title,
  CenterView,
  Description,
  Footer,
  DescriptionFooter,
  DescriptionTitle,
  WrapperDescriptions
} from './DetailsCurrencie.styles'

interface DetailsCurrencieProps {
  symbol: string
  name: string
  open: string
  close: string
  market_cap: string
  price: string
  change_percert: string
  description: string
  region: string
  currency: string
  company_name: string
}

const DetailsCurrencie: React.FC<DetailsCurrencieProps> = (
  props: DetailsCurrencieProps
) => {
  const {
    currency,
    description,
    region,
    change_percert,
    close,
    company_name,
    market_cap,
    name,
    open,
    price,
    symbol
  } = props
  return (
    <Container style={{ ...styles.shadow }}>
      <Title>{company_name}</Title>
      <Title>R$ = {price}</Title>
      <CenterView>
        <WrapperDescriptions>
          <Description>{name}</Description>
          <Description>{symbol}</Description>
        </WrapperDescriptions>

        <WrapperDescriptions>
          <View>
            <DescriptionTitle>Moeda</DescriptionTitle>
            <Description>{currency}</Description>
          </View>
          <View style={{ alignItems: 'flex-end' }}>
            <DescriptionTitle>Regi√£o</DescriptionTitle>
            <Description>{region}</Description>
          </View>
        </WrapperDescriptions>

        <WrapperDescriptions>
          <View>
            <DescriptionTitle>Abertura</DescriptionTitle>
            <Description>{close}h</Description>
          </View>
          <View style={{ alignItems: 'flex-end' }}>
            <DescriptionTitle>Fechamento</DescriptionTitle>
            <Description>{open}h</Description>
          </View>
        </WrapperDescriptions>

        <WrapperDescriptions>
          <View>
            <DescriptionTitle>Percentual</DescriptionTitle>
            <Description>{change_percert}%</Description>
          </View>
          <View style={{ alignItems: 'flex-end' }}>
            <DescriptionTitle>Valor de Mercado</DescriptionTitle>
            <Description>R$ = {market_cap}</Description>
          </View>
        </WrapperDescriptions>
      </CenterView>
      <Footer>
        <DescriptionFooter>{description}</DescriptionFooter>
      </Footer>
    </Container>
  )
}
const DetailsCurrencieMemo = React.memo(DetailsCurrencie)
export { DetailsCurrencieMemo as DetailsCurrencie }

const styles = StyleSheet.create({
  shadow: {
    shadowColor: '#27ae60',
    shadowOffset: {
      width: 0,
      height: 4
    },
    shadowOpacity: 0.35,
    shadowRadius: 4.65,

    elevation: 8
  }
})
