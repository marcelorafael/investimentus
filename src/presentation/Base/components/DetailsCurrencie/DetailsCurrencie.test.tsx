import React from 'react'

import renderer from 'react-test-renderer'
import { DetailsCurrencie } from './DetailsCurrencie'

describe('Base/DetailsCurrencie', () => {
  it('should match snapshot', () => {
    const sut = (
      <DetailsCurrencie
        name=""
        change_percert=""
        close=""
        company_name=""
        currency=""
        description=""
        market_cap=""
        open=""
        price=""
        region=""
        symbol=""
      />
    )

    const render = renderer.create(sut)

    expect(render).toMatchSnapshot()
  })
})
