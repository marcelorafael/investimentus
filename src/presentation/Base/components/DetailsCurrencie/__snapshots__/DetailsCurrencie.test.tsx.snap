// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/DetailsCurrencie should match snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4286537200,
      4280028760,
      4278190080,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 5,
        "maxHeight": 500,
        "maxWidth": 400,
        "paddingTop": 7,
        "width": 300,
      },
      Object {
        "elevation": 8,
        "shadowColor": "#27ae60",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 4.65,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "textAlign": "center",
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "textAlign": "center",
        },
      ]
    }
  >
    R$ = 
    
  </Text>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 20,
          "paddingRight": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 7,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 7,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Moeda
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Regi√£o
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 7,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Abertura
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          
          h
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Fechamento
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          
          h
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 7,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Percentual
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          
          %
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          Valor de Mercado
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          R$ = 
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "maxHeight": 200,
          "minHeight": 80,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
          },
        ]
      }
    >
      
    </Text>
  </View>
</BVLinearGradient>
`
